src/dsync/dsync.c

```diff
@@ -1884,6 +2358,10 @@ static void dsync_option_fini(void)
         dsync_output_free(output);
     }
     assert(list_empty(&options.outputs));
+
+    if (options.link_dest != NULL) {
+        mfu_free(&options.link_dest);
```

Comment:
It's safe to call mfu_free without checking that the item is NULL, because mfu_free will do that check itself.  To simplify the code, let's remove the if check here.

Reply:
Done